service/src/main/java/com/epam/rft/atsy/service/impl/ChannelServiceImpl.java

```diff
@@ -68,6 +68,12 @@ public Long saveOrUpdate(ChannelDTO channel) {
       throw new DuplicateChannelException(channel.getName());
     }
 
-    return this.channelRepository.saveAndFlush(channelEntity).getId();
+    ChannelEntity savedEntity = this.channelRepository.saveAndFlush(channelEntity);
+
+    try {
+      return savedEntity.getId();
+    } catch (NullPointerException ex) {
+      return -1L;
```

Comment:
For me it seems like a black magic. Why does this method return -1 if null pointer exception is occurred? In my opinion it is more readable if throw illegal argument exception if a validation fails or return null if savedEntity is null, but I prefer my first option. And one more question: Could you please describe a situation when savedEntity can be null?